dspace-api/src/main/java/org/dspace/external/provider/impl/SHERPAv2JournalDataProvider.java

```diff
@@ -0,0 +1,358 @@
+            log.error("SHERPA/RoMEO query failed: ", e);
+            return null;
+        } finally {
+            if (get != null) {
+                get.releaseConnection();
+            }
+        }
+        return Collections.emptyList();
+    }
+
+    /**
+     * Construct HTTP GET object for a "title contains word" query using default start/limit parameters
+     * @param query a word or phrase to search for in journal title
+     * @return HttpGet method which can then be executed by the client
+     * @throws URISyntaxException if the URL build fails
+     */
+    private HttpGet constructHttpGet(String query) throws URISyntaxException {
+        return constructHttpGet(query, -1, -1);
+    }
+
+    /**
+     * Construct HTTP GET object for a "title contains word" query
+     * @param query the search query
+     * @param start row offset
+     * @param limit number of results to return
+     * @return HttpGet object to be executed by the client
+     * @throws URISyntaxException
+     */
+    private HttpGet constructHttpGet(String query, int start, int limit) throws URISyntaxException {
+        // Build URL based on search query
+        URIBuilder uriBuilder = new URIBuilder(url);
+        uriBuilder.addParameter("item-type", "publication");
+        uriBuilder.addParameter("filter", "[[\"title\",\"contains word\",\"" + query + "\"]]");
+        uriBuilder.addParameter("format", "Json");
+        // Set optional start (offset) and limit parameters
+        if (start >= 0) {
+            uriBuilder.addParameter("offset", String.valueOf(start));
+        }
+        if (limit > 0) {
+            uriBuilder.addParameter("limit", String.valueOf(limit));
+        }
+        if (StringUtils.isNotBlank(apiKey)) {
+            uriBuilder.addParameter("api-key", apiKey);
+        }
+
+        // Create HTTP GET object
+        HttpGet method = new HttpGet(uriBuilder.build());
+
+        // Set connection parameters
+        int timeout = 5000;
+        method.setConfig(RequestConfig.custom()
+            .setConnectionRequestTimeout(timeout)
+            .setConnectTimeout(timeout)
+            .setSocketTimeout(timeout)
+            .build());
+        return new HttpGet(uriBuilder.build());
+    }
+
+    @Override
+    public boolean supports(String source) {
+        return StringUtils.equalsIgnoreCase(sourceIdentifier, source);
+    }
+
+    /**
+     * Get number of results returned from a SHERPA v2 publication search
+     * @param query The query to be search on and give the total amount of results
+     * @return int representing number of journal results
+     */
+    @Override
+    public int getNumberOfResults(String query) {
+        HttpGet get = null;
+        try {
+            get = constructHttpGet(query);
+            HttpClient hc = new DefaultHttpClient();
+            HttpResponse response = hc.execute(get);
+            if (response.getStatusLine().getStatusCode() == 200) {
```

Comment:
Can we use "HttpStatus.SC_OK" instead of 200 ?

Reply:
Done! See commit# 839159b:
https://github.com/DSpace/DSpace/pull/2773/commits/839159bc8511dff8e0b8b58eb24b60957a5bd7c0